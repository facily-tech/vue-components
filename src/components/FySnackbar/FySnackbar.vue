<template>
  <div>
    <fy-snackbar-base
      v-if="snackbar"
      v-model="snackbar.model"
      :timeout="snackbar.timeout"
      :color="snackbar.bgColor"
    >
      <p :style="`color: ${snackbar.fontColor}`" class="mb-0">{{ snackbar.text }}</p>

      <template v-slot:action="{ attrs }">
        <v-btn color="#ffffff" text v-bind="attrs" @click="snackbarClose" icon>
          <v-icon> mdi-close </v-icon>
        </v-btn>
      </template>
    </fy-snackbar-base>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { VSnackbar } from 'vuetify/lib';

import { ISnackbarProps } from './types';

const BaseVSnackbar = Vue.extend({ mixins: [VSnackbar] });

interface options extends InstanceType<typeof BaseVSnackbar> {
  snackbar: ISnackbarProps;
}

export default BaseVSnackbar.extend<options>().extend({
  name: 'fy-snacbar',

  components: { FySnackbarBase: () => import('./FySnackbarBase') },

  props: {
    snackbar: {
      type: Object as () => ISnackbarProps,
      default: {} as ISnackbarProps,
    },
  },

  methods: {
    snackbarClose(): void {
      this.$emit('snackbar-close');
    },
  },
});
</script>
