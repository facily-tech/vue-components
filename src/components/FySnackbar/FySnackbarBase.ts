/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VSnackbar } from 'vuetify/lib';

const BaseSnackbar = Vue.extend({ mixins: [VSnackbar] });

type options = InstanceType<typeof BaseSnackbar>;

export default BaseSnackbar.extend<options>().extend({
  name: 'fy-snackbar-base',
  computed: {
    class() {
      return {
        ...VSnackbar.options.computed.classes.call(this),
        'fy-input': true,
      };
    },
  },
});
