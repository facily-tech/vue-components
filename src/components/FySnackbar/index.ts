export { default as FySnackbar } from './FySnackbar.vue';
