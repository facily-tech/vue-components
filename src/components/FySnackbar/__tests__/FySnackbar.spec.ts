import Vue from 'vue';
import Vuetify from 'vuetify';
import FySnackbar from '../FySnackbar.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FySnackbar.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FySnackbar>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FySnackbar, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
