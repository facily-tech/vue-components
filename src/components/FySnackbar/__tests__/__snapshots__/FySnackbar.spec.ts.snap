// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FySnackbar.ts should render component and match snapshot 1`] = `
<div class="v-snack v-snack--bottom v-snack--has-background" snackbar="[object Object]">
  <transition-stub name="v-snack-transition">
    <div class="v-snack__wrapper v-sheet theme--dark" style="display: none;">
      <div role="status" aria-live="polite" class="v-snack__content">
        <p class="mb-0"></p>
      </div>
      <div class="v-snack__action "><button type="button" class="v-btn v-btn--icon v-btn--round v-btn--text theme--dark elevation-0 v-size--default v-snack__btn" style="color: rgb(255, 255, 255); caret-color: #ffffff;" data-test="btn" color="#ffffff" text="" icon=""><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--dark"></i></span></button></div>
    </div>
  </transition-stub>
</div>
`;
